keanu-project/src/main/java/io/improbable/keanu/templating/Sequence.java

```diff
@@ -11,18 +11,34 @@
 
 public class Sequence implements Iterable<SequenceItem> {
 
-    private List<SequenceItem> containedItems;
+    private final ArrayList<SequenceItem> containedItems;
+    private final int uniqueIdentifier;
+    private final String name;
 
-    public Sequence(int reservedSize) {
+    public Sequence(int reservedSize, int uniqueIdentifier, String name) {
         containedItems = new ArrayList<>(reservedSize);
+        this.uniqueIdentifier = uniqueIdentifier;
+        this.name = name;
+    }
+
+    public int getUniqueIdentifier() {
+        return this.uniqueIdentifier;
+    }
+
+    public String getName() {
+        return this.name;
     }
 
     public int size() {
         return containedItems.size();
     }
 
     public void add(SequenceItem p) {
-        containedItems.add(p);
+        int index = p.getIndex();
+        while (containedItems.size() <= index) {
+            containedItems.add(null);
+        }
```

Comment:
Ah yes I was looking for a method like that but couldn't find anything. Thanks :)
The reason I've done this is because the loader logic may receive vertices from one sequence item before ones from others so this was the easiest way to make sure that when I added sequence items, I wasn't having to ensure I was adding them in order. The alternative is creating a separate `put` method or building up a cache of sequenceitems before adding to the sequence but they seemed less elegant?

Reply:
Oh actually `ensureCapacity` still wont work because it wont initialise null values and allow the use of `put`. You have to use `add` which would mean using it in order

Reply:
In terms of the sequence items - will that really happen?  Wouldn't the network you save always have lower sequence items come first?  And shouldn't they always go up sequentially?

Reply:
I found this not to be the case when I called `network.getAllVertices()` the vertices weren't in sequenceItem-order so this was breaking

Reply:
Yeah, the getAllVertices is just a set, so you'll get a "random" order - that's why I was discussing the Topological Sorter as maybe being useful - then you should get your sequence items sequentially in order?

Reply:
yeah so I think this would help but I don't think it's a guarantee. If, say, you initialise a constant vertex in every sequence item then it will not have any parents and could come first in the sort. 
E.g. 
```python
def factory(sequence_item):
    two = Const(2.0)
    sequence_item.add(two)
```
In the above you could get all the `Sequence_Item_X.two`s in any order.